### DISCUSSION TITLE
Use the node-linker=hoisted, depend on not ascend

### DISCUSSION THREAD

**Discussion by yuhang1995** [pnpm/pnpm#5470] [Q&A]
In the workspace, I use the node-linker=hoisted, I hope is: all versions of the same dependence will be promoted to the top in node_modules; But the resulting dependencies are not promoted to the top-level node_modules, even though they are of the same version.

I don't know if the description is detailed, but here is a simple example: https://github.com/yuhang1995/my-turborepo

**Comment by zkochan** [comment]
I don't know what you mean. Everything looks correct in that project.

**Comment by yuhang1995** [comment]
Sorry, I didn't describe my question clearly, I use the node-linker=hoisted is trying from the other direction to solve my problems; I found a and I meet the same problem discussion, there is a more detailed [discussion](https://github.com/pnpm/pnpm/issues/5351) here, but so far no results.

**Comment by yuhang1995** [comment]
https://github.com/pnpm/pnpm/issues/5351#issuecomment-1475961408
